### PR TITLE
S1task1

### PR DESCRIPTION
**Pull request by HarryCusworth**
so i have empty test pages.
first pass at css styling
ultimate goal is to make the completed/owned things into radio buttons.

The first function extracts the names of the various campaign names

the second function sorts and displays the scenarios as grouped by those campaign names

### CODE REVIEW

File: collection_main.php [hunk 1]
```diff
@@ -1 +1,56 @@
...
+    $db = new PDO('mysql:host=db; dbname=arkham_lcg_scenarios', 'root', 'password');
+    $campaigns = $db->prepare("SELECT DISTINCT `cycle` FROM `scenarios`;");
+    $campaigns->setFetchMode(PDO::FETCH_ASSOC);
+    $campaigns->execute();
+    $cycleArray = array();
+    $results = $campaigns->fetchAll();
+    foreach ($results as $cycleName) {
+        array_push($cycleArray, $cycleName["cycle"]);
+    }
+//    var_dump($cycleArray);
```
Comment (damienscarpellini) [inline collection_main.php]: shouldn't have commented code in a repo

File: collection_main.php [hunk 2]
```diff
@@ -1 +1,56 @@
...
+        $query = "SELECT *  FROM `scenarios` WHERE `cycle` = :cycle ORDER BY `position`;";
+        $campaign = $db->prepare($query);
+        $params = [':cycle' => $cycle];
+        $campaign->setFetchMode(PDO::FETCH_ASSOC);
+        $campaign->execute($params);
+
+        $results = $campaign->fetchAll();
+        echo "brake point?<br><br>";
+        foreach ($results as $scenario) {
+            echo "<br>".var_dump($scenario)."<br>";
```
Comment (damienscarpellini) [inline collection_main.php]: same here, i think your just checking it works with the dump, however you have submitted the pull request with it still live in the code

File: collection_main.php [hunk 3]
```diff
@@ -1 +1,56 @@
...
+
+        $db = new PDO('mysql:host=db; dbname=arkham_lcg_scenarios', 'root', 'password');
+        $query = "SELECT *  FROM `scenarios` WHERE `cycle` = :cycle ORDER BY `position`;";
+        $campaign = $db->prepare($query);
+        $params = [':cycle' => $cycle];
+        $campaign->setFetchMode(PDO::FETCH_ASSOC);
+        $campaign->execute($params);
+
+        $results = $campaign->fetchAll();
+        echo "brake point?<br><br>";
```
Comment (damienscarpellini) [inline collection_main.php]: i think your just looking for positioning here. i wouldn't put echo or `<br>`  in a function.

**Review by damienscarpellini — APPROVED** [review]
looks good so far.

I would recommend using a functions page and linking that to the other pages to make testing easy.

File: main_stylesheet.css [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+    display: flex;
+    flex-direction: column;
+    justify-items: left;
+
+}
+
+.flexContainer {
+    display: flex;
+    flex-direction: row;
+    justify-content: space-between;
```
Comment (waseembrare) [inline main_stylesheet.css:55]: - when resizing down to around mobile width the child containers outgrow the parent cycles box. I think the fix might be overflow: auto might be the fix (i hate flex, not the best person to diagnose) https://stackoverflow.com/questions/41674979/flex-child-is-growing-out-of-parent

File: main_stylesheet.css [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+    text-decoration: none;
+}
+
+a:link, a:visited {
+    color: #CE3B45;
+}
+
+a:hover {
+    color: #215097;
+}
```
Comment (waseembrare) [inline main_stylesheet.css]: - Can see some redundant selectors used like a for example where no a tags within php file. If you gonna be putting them in and styled ahead of time just ignore.

File: arkham_lcg_scenarios_2021-02-08.sql [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+
+DROP TABLE IF EXISTS `scenarios`;
+
+CREATE TABLE `scenarios` (
+  `id` int(11) unsigned NOT NULL AUTO_INCREMENT,
+  `name` varchar(50) DEFAULT NULL,
+  `cycle` varchar(50) DEFAULT NULL,
+  `position` tinyint(2) DEFAULT NULL,
+  `standalone` tinyint(1) DEFAULT NULL,
+  `owned` tinyint(1) DEFAULT NULL,
```
Comment (waseembrare) [inline arkham_lcg_scenarios_2021-02-08.sql:34]: - for the binary fields you could use 'unsigned' checkbox in table structure tab to ensure no minus figures can be entered. I guess you won't allow user to input minus as you'll prob use a checkbox but thought so it's likely not an issue

**Review by waseembrare — COMMENTED** [review]
- Looks great on the whole, really like the simplicity of the code and the functionality of the app. 
- Not much to comment on, syntax is looking neat, makes sense to read.

File: main_stylesheet.css (hunk 4 above)
Comment (HarryCusworth) [reply main_stylesheet.css:55]: fixed by turning of flex in mobile view

File: arkham_lcg_scenarios_2021-02-08.sql (hunk 6 above)
Comment (HarryCusworth) [reply arkham_lcg_scenarios_2021-02-08.sql:34]: I think tiny int is the most stable way of doing boolean in SQL as check box isn't always supported

File: main_stylesheet.css (hunk 5 above)
Comment (HarryCusworth) [reply main_stylesheet.css]: yeah just left in for future links

**Comment by HarryCusworth** [comment]
Awaiting verifiaction on unit testing otherwise all reviews and edits made

**Pull request merged**